tests/GhcSession.hs

```diff
@@ -97,41 +100,56 @@ compileModule ep opts = do
 
     E.handle (\(ec :: ExitCode) -> print ec >> return False) $ do
 
-    defaultErrorHandler defaultFatalMessager defaultFlushOut $ do
-
-    runGhc (Just libdir) $ do
-
-    dflags0 <- getSessionDynFlags
-    let dflags1 = dflags0 {
-        ghcMode   = CompManager
-      , ghcLink   = LinkInMemory
-      , hscTarget = HscInterpreted
-      , optLevel  = 0
-      }
-
-    (dflags2, _, _) <- parseDynamicFlags dflags1 (map noLoc opts)
-    _ <- setSessionDynFlags dflags2
-
-    ts <- mapM (\t -> guessTarget t Nothing) $
-         case ep of
-           ChLibEntrypoint ms ms' -> map unChModuleName $ ms ++ ms'
-           ChExeEntrypoint m  ms  -> [m] ++ map unChModuleName ms
-           ChSetupEntrypoint      -> ["Setup.hs"]
-    let ts' = map (\t -> t { targetAllowObjCode = False }) ts
-
-    setTargets ts'
-    _ <- load LoadAllTargets
-
-    setContext $ case ep of
-      ChLibEntrypoint ms ms' ->
-          map (IIModule . mkModuleName . unChModuleName) $ ms ++ ms'
-      ChExeEntrypoint _  ms  ->
-          map (IIModule . mkModuleName . unChModuleName) $ ChModuleName "Main" : ms
-      ChSetupEntrypoint      ->
-          map (IIModule . mkModuleName) ["Main"]
-
-    liftIO $ print ExitSuccess
-    return True
+#if __GLASGOW_HASKELL__ <= 704
+      defaultErrorHandler defaultLogAction $ do
+#else
+      defaultErrorHandler defaultFatalMessager defaultFlushOut $ do
+#endif
+
+        runGhc (Just libdir) $ do
```

Comment:
Please undo the re-indent in this chunk, makes it impossible to review. Note that `NondecreasingIndentation` is on by default.